### PR TITLE
Resolved few bugs in MainUI for User Login System

### PR DESCRIPTION
**Pull request by happynasit**
I have implemented the use cases of the User feature - CreateUser, LoginUser and DeleteUser. All three of them follow Clean architecture and the SOLID Principle.
CreateUser - functionality that creates and adds the user to the database
LoginUser - functionality that checks the existence of the user in the Database
DeleteUser - functionality that deletes the user from the database

I have covered 34% test coverage and I will be working on the test cases for use cases and the controller. 
I have also successfully implemented the CreateUserUI, LoginUserUI, and the Home Screen with with the implementation of the Action Listener.

### CODE REVIEW

File: src/main/java/use_cases/user/create_user/CreateUserInteractor.java [hunk 1]
```diff
@@ -29,46 +25,35 @@ public CreateUserInteractor(CreateUserOutputBoundary outputBoundary, CreateUserD
...
         if (!user.passwordStrength()){
             // if the given password is not strong enough then it shows fail view
-            CreateUserOutputData outputData = new CreateUserOutputData("Password is not strong enough " +
-                    "Please re enter the password.");
-            return outputBoundary.prepareFailView(outputData);
+            return new CreateUserOutputData(false, "The password is not strong enough");
 
         }else{
 
             // When the Account is successfully created.
```
Comment (JoAndNight) [inline src/main/java/use_cases/user/create_user/CreateUserInteractor.java:49]: We can deleted these comments. They look like code smells.

**Review by JoAndNight — APPROVED** [review]

File: src/main/java/ui/users/HomeScreenUI.java [hunk 2]
```diff
@@ -0,0 +1,49 @@
...
+import javax.swing.JButton;
+import javax.swing.JFrame;
+import javax.swing.JLabel;
+import javax.swing.JPanel;
+
+/**
+ * The Screen that pops up once the user is successfully logged in
+ * @author happynasit
+ */
+public class HomeScreenUI extends JFrame {
```
Comment (AngelEyesMagicMaker) [inline src/main/java/ui/users/HomeScreenUI.java:13]: A more meaningful way to organize UI.

File: src/main/java/ui/users/LoginUserUI.java [hunk 3]
```diff
@@ -0,0 +1,116 @@
...
+        loginButton.addActionListener(this);
+
+        JButton createAccountButton = new JButton("Create Account!");
+        createAccountButton.setBounds(140, 130, 300, 35);
+
+        ui.CreateAccountUI createAccountUI = new ui.CreateAccountUI(this.createUserController, this.gateway, this.loginUserController);
+        createAccountButton.addActionListener(e -> createAccountUI.main());
+
+
+        // creating the text fields
```
Comment (AngelEyesMagicMaker) [inline src/main/java/ui/users/LoginUserUI.java:69]: Maybe delete the comments later.

File: src/test/java/test_ui/TimerViewTest 2.java [hunk 4]
```diff
@@ -0,0 +1,7 @@
+package test_ui;
+
+import static org.junit.jupiter.api.Assertions.*;
+
+class TimerViewTest {
```
Comment (AngelEyesMagicMaker) [inline src/test/java/test_ui/TimerViewTest 2.java:5]: Add tests later in this file.

**Review by AngelEyesMagicMaker — DISMISSED** [review]

**Review by JoAndNight — DISMISSED** [review]

**Review by AngelEyesMagicMaker — APPROVED** [review]

File: src/main/java/ui/users/LoginUserUI.java [hunk 5]
```diff
@@ -0,0 +1,116 @@
...
+import java.awt.event.ActionListener;
+
+/**
+ * UI for the Login Page. By the actions of the buttons, the page will either be directed to the CreateAccountUI or the
+ * Menu Page of the User.
+ * @author happynasit
+ */
+public class LoginPageUI extends JFrame implements ActionListener{
+    static final JFrame mainframe = new JFrame();
+    final LoginUserController loginUserController;
```
Comment (EmilyDo) [inline src/main/java/ui/users/LoginUserUI.java:21]: scope of the variables?

File: src/main/java/use_cases/user/create_user/CreateUserInteractor.java [hunk 6]
```diff
@@ -29,46 +25,35 @@ public CreateUserInteractor(CreateUserOutputBoundary outputBoundary, CreateUserD
 
     @Override
     public CreateUserOutputData create(CreateUserInputData inputData) {
+        if (inputData.getEmailAddress().equals("") || inputData.getPassword().equals("")
+                || inputData.getRepeatPassword().equals("") || inputData.getFirstName().equals("") || inputData.getLastName().equals("")){
```
Comment (EmilyDo) [inline src/main/java/use_cases/user/create_user/CreateUserInteractor.java:29]: think there's a safer way to check if the input is empty --> .isempty() maybe?

**Review by EmilyDo — APPROVED** [review]

**Review by EmilyDo — APPROVED** [review]

**Review by JoAndNight — APPROVED** [review]

**Pull request merged**